aikau/src/main/resources/alfresco/renderers/_ActionsMixin.js

```diff
@@ -121,19 +124,33 @@ define(["dojo/_base/declare",
        * @param {object} action The configuration for the action to add
        * @param (integer} index The index of the action
        */
-      addAction: function alfresco_renderers__ActionsMixin__addAction(action, index) {
+      addAction: function alfresco_renderers__ActionsMixin__addAction(action, /*jshint unused:false*/index) {
          if (this.filterActions === false || AlfArray.arrayContains(this.allowedActions, action.id))
          {
             this.alfLog("log", "Adding action", action);
 
-            var payload = (action.publishPayload != null) ? action.publishPayload : {document: this.currentItem, action: action};
+            // If there is a node object then create the corresponding jsNode object - this is required for 
+            // label processing below
+            if (!this.currentItem.jsNode && this.currentItem.node)
+            {
+               this.currentItem.jsNode = new JsNode(this.currentItem.node);
+            }
+
+            // Some Share actions have variable labels, most notably the simple workflow actions, and these should
+            // be processed to ensure that they are rendered correctly.
+            if (action.label)
+            {
+               action.label = this.processTokens(action.label, this.currentItem);
+            }
+
+            var payload = (action.publishPayload) ? action.publishPayload : {document: this.currentItem, action: action};
             var menuItem = new AlfMenuItem({
                label: action.label,
                iconImage: AlfConstants.URL_RESCONTEXT + "components/documentlibrary/actions/" + action.icon + "-16.png",
                type: action.type,
                pubSubScope: this.pubSubScope,
                parentPubSubScope: this.parentPubSubScope,
-               publishTopic: (action.publishTopic != null) ? action.publishTopic : this.singleDocumentActionTopic,
+               publishTopic: (action.publishTopic) ? action.publishTopic : this.singleDocumentActionTopic,
                publishPayload: this.generatePayload(payload, this.currentItem, null, action.publishPayloadType, action.publishPayloadItemMixin, action.publishPayloadModifiers),
```

Comment:
This could be written `publishTopic: action.publishTopic || this.singleDocumentActionTopic` if you wanted